### DISCUSSION TITLE
New UI progress

### DISCUSSION THREAD

**Discussion by ccoors** [Hypfer/Valetudo#1110] [Archive]
Just my To-Do list for the new frontend

# Progress

## Done

- [x] Timers
- [x] Persistent map control
- [x] Map reset
- [x] Speaker test/volume
- [x] Events
- [x] Key lock
- [x] Carpet mode control
- [x] Obstacle Avoidance control
- [x] Auto empty dock auto control
- [x] Mapping pass
- [x] Voice pack management
- [x] DND Config
- [x] WiFi configuration
- [x] Manual control
- [x] HTTP Auth config
- [x] NTP State/Config

## Won't
- LED control (no robot implementation)
- Sensor calibration (no robot implementation)
- Map snapshot [needs to be rethought/removed](https://t.me/c/1357472831/59746)

## Can't
- Map segment create/edit/rename
- Combined virtual restrictions create/edit
- GoTo location preset create/edit/rename

# Chores

- [x] Try Masonry for robot settings

**Comment by gvolt** [comment]
Just wanted to say thank you, the new UI is awesome. You (@ccoors and @Hypfer) are on a roll currently. I appreciate your efforts very much. And after the timer configuration recently has reappeared in the UI I am back on the Valetudo "train" and made the leap from ancient 0.6.1 to 2021.09.1. Looking forward to the more and more completing UI. Keep up the great work!

P.S. Feel free to delete my post, if it should interfere with your working flow :)
